### PR TITLE
ref(html5): refactor presentation-uploader toast and modal

### PR DESCRIPTION
**Pull request by GuiLeme**
### What does this PR do?

This PR refactor both the presentation-uploader modal and toast to enhance stability due to some recent minor issues.

In total it:
- Updates react-toastify js library for version `11.0.2` and makes necessary changes to not break anything;
- Refactors the logic for updating the Presentation state which was confusing in some points;
- Also fix some minor lint warnings along the way;

### Motivation

Stabilize behaviour of Presentation modal and toast.

### How to test

To test, one will need to run `npm ci` to update the `node_modules` and then run the html5 service (dev or prod mode), then try as many cases as possible (try edge cases too, in order to see if something unpredicted will break as well).

### CODE REVIEW

**Comment by gustavotrott** [comment]
I noticed two differences compared to version 2.7. I know these issues aren’t introduced by your PR, so we can fix them in a separate PR if you prefer:

1. In 2.7, the “Clear errors” button didn’t depend on a “Confirm” action. In 3.0, if I click “Cancel” instead of “Confirm” when closing the modal, the errors reappear the next time.
2. In 2.7, a ✅ icon was displayed in the upload toast when the presentation finished uploading. In 3.0, there’s no immediate confirmation—it only shows an alert about the “Current presentation,” which isn’t very helpful when multiple files are uploaded.



---

- Video comparing 2.7 and 3.0 behavior:

[Screencast from 2025-01-20 09-20-01.webm](https://github.com/user-attachments/assets/31baf476-bdbd-428b-8dff-84c6e69b8782)

**Comment by gustavotrott** [comment]
When I press the button to start the upload, a toast message appears with the title ‘1 upload complete,’ even though it hasn’t started yet.

1-upload-complete

File: bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx [hunk 1]
```diff
@@ -764,7 +774,7 @@ class PresentationUploader extends Component {
     try {
       presentationsSorted = presentations
         .sort((a, b) => a.uploadTimestamp - b.uploadTimestamp)
-        .sort((a, b) => a.name.localeCompare(b.name))
+        .sort((a, b) => a.name.louacaleCompare(b.name))
```
Comment (ramonlsouza) [inline bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx:777]:
```suggestion
        .sort((a, b) => a.name.localeCompare(b.name))
```

**Comment by GuiLeme** [comment]
@gustavotrott as discussed internally, those suggestions you brought up will be implemented in a near-future PR, and are marked in this issue: https://github.com/bigbluebutton/bigbluebutton/issues/22038. Thank you for those.

**Review by ramonlsouza — APPROVED** [review]

**Review by gustavotrott — APPROVED** [review]

**Comment by gustavotrott** [comment]
Well done!!! It's reeeeally better now!
